File: summary_ranges.cpp
// 228. Summary Ranges My Submissions Question

// class Solution {
// public:
//     vector<string> summaryRanges(vector<int>& nums) {
//         vector<string> vec_str;
//         auto beg = nums.cbegin();
//         auto end = beg;
//         while (end != nums.cend()) {
//             while (end + 1 != nums.cend() && *(end + 1) - *end == 1)
//                 end++;
//             if (beg == end)
//                 vec_str.push_back(to_string(*beg));
//             else
//                 vec_str.push_back(to_string(*beg) + "->" + to_string(*end));
//             beg = end + 1;
//             end = beg;
//         }
//         return vec_str;
//     }
// };

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> vec_str;
        auto beg = 0;
        auto end = beg;
        auto size = nums.size();
        while (end < size) {
            while (end + 1 < size && nums[end + 1] - nums[end] == 1)
                end++;
            if (beg == end)
                vec_str.push_back(to_string(nums[beg]));
            else
                vec_str.push_back(to_string(nums[beg]) + "->" + to_string(nums[end]));
            beg = end + 1;
            end = beg;
        }
        return vec_str;
    }
};

File: summary_ranges.py
# 228. Summary Ranges My Submissions Question

class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        beg = 0
        end = beg
        size = len(nums)
        ret = []
        while end < size:
            while end + 1 < size and nums[end + 1] - nums[end] == 1:
                end += 1
            if beg == end:
                ret.append(str(nums[beg]))
            else:
                ret.append(str(nums[beg]) + "->" + str(nums[end]))
            beg = end + 1
            end = beg
        return ret
                    
            
                    